lib/mix/lib/mix/tasks/format.ex

```diff
@@ -177,6 +177,10 @@ defmodule Mix.Tasks.Format do
     end
 
     {not_equivalent, not_formatted}
+  rescue
+    e ->
+      Mix.shell().error("mix format failed when parsing file: #{file}")
+      reraise e, System.stacktrace()
```

Comment:
Btw, you need to read the stacktrace as the first step in a rescue. That's because if `Mix.shell().error` try/catch something, then the stacktrace value changes. But no worries, I will fix it locally!